[optim] take kw-only argument for functional optim APIs

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56185 [optim] take kw-only argument for functional optim APIs**

Differential Revision: [D27802169](https://our.internmc.facebook.com/intern/diff/D27802169/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27802169/)!

Commit:
[optim] take kw-only argument for functional optim APIs



Differential Revision: [D27802169](https://our.internmc.facebook.com/intern/diff/D27802169/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27802169/)!

[ghstack-poisoned]

albanD reviewed (approved):
Nice! that helps a lot to ensure there is no mismatch in the argument names!

Commit:
Update on "[optim] take kw-only argument for functional optim APIs"


Differential Revision: [D27802169](https://our.internmc.facebook.com/intern/diff/D27802169/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27802169/)!

[ghstack-poisoned]
